src/main/scala/fi/oph/koski/ytr/YoTodistusServlet.scala

```diff
@@ -0,0 +1,52 @@
+package fi.oph.koski.ytr
+
+import fi.oph.koski.config.KoskiApplication
+import fi.oph.koski.http.{HttpStatus, KoskiErrorCategory}
+import fi.oph.koski.koskiuser.{KoskiSpecificSession, RequiresKansalainen}
+import fi.oph.koski.log.{AuditLog, KoskiAuditLogMessage, KoskiAuditLogMessageField}
+import fi.oph.koski.log.KoskiOperation.{KoskiOperation, YTR_YOTODISTUKSEN_LUONTI, YTR_YOTODISTUKSEN_LATAAMINEN}
+import fi.oph.koski.servlet.{KoskiSpecificApiServlet, NoCache}
+
+class YoTodistusServlet(implicit val application: KoskiApplication)
+  extends KoskiSpecificApiServlet
+    with NoCache
+    with RequiresKansalainen {
+
+  val service: YoTodistusService = application.yoTodistusService
+
+  get("/status/:lang/:oppijaOid") {
+    renderEither(getRequest.flatMap(service.currentStatus))
+  }
+
+  get("/generate/:lang/:oppijaOid") {
+    mkAuditLog(session, YTR_YOTODISTUKSEN_LUONTI)
```

Comment:
Pitäisiköhän tämä auditlokitus tehdä vasta käyttöoikeustarkastuksen jälkeen? Nyt toimii ainakin eri tavalla kuin download:in audilokitus.

Reply:
Totta!